controllers/competition_supervisor/competition_supervisor.py

```diff
@@ -143,7 +143,13 @@ def wait_until_robots_ready(supervisor: Supervisor) -> None:
 
         if field.getSFString() != 'ready':
             print("Waiting for {}".format(zone_id))
+            start_time = supervisor.getTime()
             while field.getSFString() != 'ready':
+                # 5 second initialisation timeout
+                if (supervisor.getTime() - start_time) > 5:
```

Comment:
Computing the end-time upfront is usually slightly clearer:
```suggestion
            end_time = supervisor.getTime() + 5
            while field.getSFString() != 'ready':
                # 5 second initialisation timeout
                if supervisor.getTime() > end_time:
```